src/api/v1/db/oracledb/ingredients-dao.js

```diff
@@ -15,6 +17,70 @@ const ingredientsColumnNames = {
   notes: 'NOTES',
 };
 
+/**
+ * convert an output bind param name like `nameOut` to a
+ * property name like `name` by removing 'Out' at the end of the string
+ * @param {string} outBindParamName
+ * @returns {string} the property name
+ */
+const outBindParamToPropertyName = (outBindParamName) => outBindParamName.slice(0, -3);
+
+/**
+ * A list of the columns in the INGREDIENTS table joined by commas
+ * @constant
+ * @type {string}
+ */
+const ingredientColumns = _.map(ingredientsProperties,
+  (propertyValues, property) => ingredientsColumnNames[property])
+  .join(', ');
```

Comment:
You can just use [`_.keys`](https://lodash.com/docs/4.17.15#keys) to improve the logic here.

Reply:
If we used `_.keys` though, wouldn't we have to call `map` again on the result? I'm not sure that's a clear improvement over just `_.map`.

Reply:
You are right, I didn't notice you used the keys of as new keys `ingredientsProperties` of `ingredientsColumnNames`. Then yes, using `_.map` sounds like the best solution here. Good catch.